src/components/AccountLoginForm.jsx

```diff
@@ -4,17 +4,90 @@ import React from 'react'
 import classNames from 'classnames'
 import statefulForm from '../lib/statefulForm'
 import { translate } from 'cozy-ui/react/I18n'
-import Field, { PasswordField, DropdownField, CheckboxField, isHidden, isAdvanced } from './Field'
+import Field, { PasswordField, DropdownField, CheckboxField, isHidden, isAdvanced, FolderPickerField } from './Field'
 import ReactMarkdownWrapper from './ReactMarkdownWrapper'
 
-const AccountLoginForm = ({ t, isOAuth, oAuthTerminated, fields, error, dirty, submitting, forceDisabled, forceEnabled, values, submit, connectorSlug, isSuccess, disableSuccessTimeout, isUnloading }) => {
+const AccountLoginForm = ({ t, isOAuth, oAuthTerminated, fields, error, dirty, submitting, forceDisabled, forceEnabled, values, submit, connectorSlug, isSuccess, disableSuccessTimeout, isUnloading, displayAdvanced, toggleAdvanced, disableFolderPath }) => {
   const isUpdate = !!values && Object.keys(values).length > 0
   let alreadyFocused = false
   const editableFields = Object.keys(fields)
     .filter(name => !isHidden(fields[name]) && !isAdvanced(fields[name]))
     .map(name => ({ ...fields[name], name }))
+  const advancedFields = Object.keys(fields)
+    .filter(name => !isHidden(fields[name]) && isAdvanced(fields[name]))
+    .map(name => ({ ...fields[name], name }))
   const hasEditableFields = !!editableFields.length
   const submitEnabled = dirty || (isOAuth && !(isUpdate && hasEditableFields)) || forceEnabled
+
+  const renderField = (field) => {
+    const { name } = field
+    const readOnly = name === 'login' && isUpdate
+    const giveFocus = !alreadyFocused && !readOnly
+    if (giveFocus) alreadyFocused = giveFocus
+    const inputName = `${name}_${connectorSlug}`
+    const description = field.hasDescription
+      ? <ReactMarkdownWrapper source={t(`connector.${connectorSlug}.description.field.${name}`)} />
+      : ''
+    const onEnterKey = () => {
+      if (((!isUpdate || hasEditableFields) && !isSuccess) && !submitting && submitEnabled) submit()
+    }
+    switch (field.type) {
```

Comment:
This pattern switch / case pattern will be probably use for other form, it could be a good idea in the future to have a global form lib we can import easily in whatever component (probably a job for cozy-ui ?)

Reply:
Completely agree even it's not the priority for now. Collect has the most complicated but it could be a good idea to have it for other applications 👍